src/locale/keymap-util.c

```diff
@@ -314,15 +375,24 @@ int vconsole_write_data(Context *c) {
                 if (unlink("/etc/vconsole.conf") < 0)
                         return errno == ENOENT ? 0 : -errno;
 
+                c->vc_mtime = USEC_INFINITY;
                 return 0;
         }
 
-        return write_env_file_label("/etc/vconsole.conf", l);
+        r = write_env_file_label("/etc/vconsole.conf", l);
+        if (r < 0)
+                return r;
+
+        if (stat("/etc/vconsole.conf", &st) >= 0)
+                c->vc_mtime = timespec_load(&st.st_mtim);
```

Comment:
hmm, i'd prefer if we'd actually read the mtime from the file we just wrote as last step before closing its fd, otherwise this has such a "racy" feeling to it... I figure doing that would not be trivial, hence maybe it is OK this way, but it sure would be prettier if we somehow made this work one day (for example, have write_env_file_label() and friends optionally return the final mtime after it completed writing)